File: Lista1/src/exec4_1.java
/**
 *
 * @author 11800910
 */
public class exec4_1 {

    public static void main(String[] args) {
        System.out.println("EXERCICIO 4 (parte 1): Arvore direita->esquerda\n");
        int num = 0;
        boolean leu = false;
        while(!leu){
            leu = true;
            System.out.printf("Digite um inteiro: ");
            try {
                num = EntradaTeclado.leInt();
                if( num < 0){
                    System.out.println("Número deve ser positivo!!");
                    leu = false;
                }
            } catch (Exception e) {
            System.out.println("O valor que vc digitou não é um número inteiro");
                leu = false;
            }
        }
        // com num validado, vamos montar a arvore
        int asterisco = num;
        int espaco = 0;
        while(asterisco > 0){
            for(int i = 0; i < espaco; i++)
                System.out.printf(" ");
            for(int i = 0; i < asterisco; i++)
                System.out.printf("*");
            System.out.printf("\n");
            asterisco--;
            espaco++;
        }
    }

}

File: Lista2/exec4_1.py
'''
 *
 * @author 11800910
'''

def valida_entrada():
    while(True):
        try:
            num = int(input('Digite um inteiro: '))
            if num < 0:
                print('Número deve ser positivo!!')
                continue
            break
        except ValueError:
            print('O valor que vc digitou não é um número inteiro')
    return num

if __name__ == '__main__':

    # cabeçalho
    print('EXERCICIO 4 (parte 1): Arvore direita->esquerda\n')
    
    # valdação da entrada 
    num = valida_entrada()

    # com num validado, vamos montar a arvore
    asteriscos = num
    espacos = 0
    # loop vertical (até acabar o num)
    while asteriscos > 0:
        # loop horizontal (até acabar a linha)
        for _ in range(0, espacos):
            print(' ', end="")
        # espacos roda horizontalmente de 0 até num
        for _ in range(0, asteriscos):
            print('*', end="")
        print('\n')
        asteriscos -= 1
        espacos += 1